### ISSUE TITLE
Add function to delete internal context.data

### ISSUE THREAD

**Issue by byanjiong** [go-ozzo/ozzo-routing#55]
We use `c.Set("isUser", true)` to pass data to other handlers, sometimes in complex functions, we may use a few handlers inside a handler, some handlers may required some internal data to be set while other handlers do not need those data, so hope to have a function to "delete" the context.data

```
func abc123(c *routing.Context) error {

    c.Set("name1", true)
    err := cde123(c)

    c.Delete("name1") // <-- hope to have something like that to delete internal data
    err := cde123(c)

    return nil
}
```

**Comment by kolkov** [comment]
Makes sense — being able to clean up context data between handler chains is a reasonable request. Currently there's no `Delete` method on the context, but as a workaround you can `c.Set("name1", nil)` to effectively clear a value.

This issue has been sitting for a long time though. Is this still relevant for you? We'll close it if there's no response in a couple of weeks.
